### PR TITLE
[MM-20329] UI Automation: "Can change Terms of Service Link setting"

### PR DESCRIPTION
**Pull request by kosgrz**
#### Summary
Write an automated test using Cypress for "Can change Terms of Service Link setting"

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13167

### CODE REVIEW

File: e2e/cypress/integration/system_console/ui_and_api/customization_spec.js [hunk 1]
```diff
@@ -54,4 +54,35 @@ describe('Customization', () => {
...
+        cy.findByTestId('SupportSettings.TermsOfServiceLinkhelp-text').find('span').should('be.visible').and('have.text',
+            'Link to the terms under which users may use your online service. By default, this includes the ' +
+            '"Mattermost Conditions of Use (End Users)" explaining the terms under which Mattermost software is ' +
+            'provided to end users. If you change the default link to add your own terms for using the service you ' +
+            'provide, your new terms must include a link to the default terms so end users are aware of the Mattermost ' +
+            'Conditions of Use (End User) for Mattermost software.');
+
+        // # Generate and enter a random help link
+        const newValue = 'https://test.com';
+        cy.findByTestId('TeamSettings.SiteNameinput').clear().type(newValue);
```
Comment (saturninoabril) [inline e2e/cypress/integration/system_console/ui_and_api/customization_spec.js]: I think this should be
```suggestion
        cy.findByTestId('SupportSettings.TermsOfServiceLinkinput').clear().type(newValue);
```

File: e2e/cypress/integration/system_console/ui_and_api/customization_spec.js [hunk 2]
```diff
@@ -54,4 +54,35 @@ describe('Customization', () => {
...
+
+        // * Verify the site name's help text is visible and matches the text
+        cy.findByTestId('SupportSettings.TermsOfServiceLinkhelp-text').find('span').should('be.visible').and('have.text',
+            'Link to the terms under which users may use your online service. By default, this includes the ' +
+            '"Mattermost Conditions of Use (End Users)" explaining the terms under which Mattermost software is ' +
+            'provided to end users. If you change the default link to add your own terms for using the service you ' +
+            'provide, your new terms must include a link to the default terms so end users are aware of the Mattermost ' +
+            'Conditions of Use (End User) for Mattermost software.');
+
+        // # Generate and enter a random help link
```
Comment (saturninoabril) [inline e2e/cypress/integration/system_console/ui_and_api/customization_spec.js]: This is not accurate (also not random).  Please update.

File: e2e/cypress/integration/system_console/ui_and_api/customization_spec.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ describe('Customization', () => {
         cy.apiGetConfig().then((response) => {
             const config = response.body;
             origConfig = {
-                SupportSettings: {HelpLink: config.SupportSettings.HelpLink},
+                SupportSettings: {HelpLink: config.SupportSettings.HelpLink, TermsOfServiceLink: config.SupportSettings.TermsOfServiceLink},
```
Comment (saturninoabril) [inline e2e/cypress/integration/system_console/ui_and_api/customization_spec.js]: Please break into multiple lines.

File: e2e/cypress/integration/system_console/ui_and_api/customization_spec.js (hunk 1 above)
Comment (kosgrz) [reply e2e/cypress/integration/system_console/ui_and_api/customization_spec.js]: Such a stupid mistake :/. Done :+1:

File: e2e/cypress/integration/system_console/ui_and_api/customization_spec.js (hunk 2 above)
Comment (kosgrz) [reply e2e/cypress/integration/system_console/ui_and_api/customization_spec.js]: Done :+1:

File: e2e/cypress/integration/system_console/ui_and_api/customization_spec.js (hunk 3 above)
Comment (kosgrz) [reply e2e/cypress/integration/system_console/ui_and_api/customization_spec.js]: Done :+1:

File: e2e/cypress/integration/system_console/ui_and_api/customization_spec.js [hunk 4]
```diff
@@ -219,4 +220,35 @@ describe('Customization', () => {
...
+            '"Mattermost Conditions of Use (End Users)" explaining the terms under which Mattermost software is ' +
+            'provided to end users. If you change the default link to add your own terms for using the service you ' +
+            'provide, your new terms must include a link to the default terms so end users are aware of the Mattermost ' +
+            'Conditions of Use (End User) for Mattermost software.');
+
+        // # Enter a new help link
+        const newValue = 'https://test.com';
+        cy.findByTestId('SupportSettings.TermsOfServiceLinkinput').clear().type(newValue);
+
+        // # Click Save button
```
Comment (hmhealey) [inline e2e/cypress/integration/system_console/ui_and_api/customization_spec.js]: What's the meaning of the `*` and `#` in these comments?

Comment (saturninoabril) [reply e2e/cypress/integration/system_console/ui_and_api/customization_spec.js]: `*` indicates verification/assertion and `#` as test steps. It's a style guide [normally found on header of each spec file](https://github.com/mattermost/mattermost-webapp/pull/4370/files#diff-e8e5668a9849acec0e1a98afb29a0883L4-L8) that helps non-tech QA understand the test flow without digging the codes and be able to compare whether the written code represents the written test case.

Comment (hmhealey) [reply e2e/cypress/integration/system_console/ui_and_api/customization_spec.js]: That makes sense. I haven't seen those headers at the tops of files. I'll make sure to get people to add those when reviewing E2E tests in the future

**Comment by saturninoabril** [comment]
@kosgrz Please update and merge conflicts and then good to merge, thanks

**Pull request merged**